pkg/operator/ceph/object/objectstore.go

```diff
@@ -1010,3 +1017,17 @@ func errorOrIsNotFound(err error, msg string, args ...string) error {
 	}
 	return errors.Wrapf(err, msg, args)
 }
+
+// Update the Endpoints in Zone with new details and order of the list does not matter
+func updateZoneEndpointList(zoneEndpointsList []string, undesiredEndpoint, desiredEndpoint string) []string {
```

Comment:
It would be good to update the comment here with some expectations about behavior for understanding how the function can/should be used.
- "desiredEndpoint is given a higher precedence than undesiredEndpoint; if both are the same, the function keeps desiredEndpoint in the resulting list" is a critical detail
- "if no desiredEndpoint is given, the undesiredEndpoint is still removed" is another
- "if no undesiredEndpoint is given, the desiredEndpoint is still added" similarly